clang/tools/driver/cc1_main.cpp

```diff
@@ -311,9 +372,187 @@ class raw_mirroring_ostream : public llvm::raw_ostream {
 };
 } // namespace
 
+Optional<int> CompileJobCache::tryReplayCachedResult(CompilerInstance &Clang) {
+  if (!CacheCompileJob)
+    return None;
+
+  // Create the result cache key once Invocation has been canonicalized.
+  ResultCacheKey =
+      createResultCacheKey(*CAS, Clang.getDiagnostics(), Clang.getInvocation());
+  if (!ResultCacheKey)
+    return 1;
+
+  Expected<llvm::cas::CASID> Result = CAS->getCachedResult(*ResultCacheKey);
```

Comment:
Just a "drive-by" comment, not related to this particular PR, but `CAS->getCachedResult()` should really be returning `Expected<Optional<llvm::cas::CASID>>` and not conflate "missing value" with "error occurred".

Reply:
Agreed; this was a very poor choice. There are FIXMEs (somewhere) about this already :).